eng/Analyzers.targets

```diff
@@ -14,7 +14,7 @@
   <PropertyGroup Condition="'$(RunAnalyzers)' != 'false'">
     <EnableSingleFileAnalyzer Condition="
       '$(EnableSingleFileAnalyzer)' == '' And
-      '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</EnableSingleFileAnalyzer>
+      $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableSingleFileAnalyzer>
```

Comment:
I don't understand this change. dotnet/runtime main only builds net8.0 and net9.0. Why do we need this check?

Reply:
Are you sure? I see a lot in Microsoft.Extensions that has netstandard2.0

Reply:
Exactly. The `TFI == .NETCoreApp` check should already take care of only applying to .NETCoreApp TFMs.